lib/main.rb

```diff
@@ -0,0 +1,184 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    return to_enum(:each) unless block_given?
+
+    i = 0
+    while i < size
+      yield(self[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(:each) unless block_given?
+
+    i = 0
+    while i < size
+      yield(self[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    return to_enum(:each) unless block_given?
+
+    new_arr = []
+    my_each do |x|
+      new_arr << x if yield(x) == true
+    end
+    new_arr
+  end
+
+  def my_any?(arg = nil)
+    count = []
+    if arg.nil?
+      if block_given?
+        my_each do |x|
+          count << true if yield(x) == true
+        end
+      else
+        count = my_select { |x| x != false && !x.nil? }
+      end
+    elsif arg.class == Class
+      count = my_select { |x| x.is_a?(arg) }
+    elsif arg.class == Regexp
+      my_each do |x|
+        return true if x.match(arg)
+      end
+    else
+      count = my_select { |x| x == arg }
+    end
+    p count
+    count.size.positive?
+  end
+
+  def my_all?(arg = nil)
+    unless block_given?
+      falsy = 0
+      if arg.nil?
+        my_each do |x|
+          falsy += 1 if x == false || x.nil?
+        end
+      elsif arg.class == Class
+        my_each do |x|
+          falsy += 1 unless x.is_a?(arg)
+        end
+      elsif arg.class == Regexp
+        num = 0
+        my_each do |x|
+          num += 1 if x.match(arg)
+        end
+        return num == size
+      elsif arg.class == Integer
+        return self == [arg]
+      end
+      return falsy.zero?
+    end
+
+    pos = 0
+    my_each do |x|
+      pos += 1 if yield(x) == true
+    end
+    pos == size
+  end
+
+  def my_none?(arg = nil)
+    count = []
+    if block_given?
+      my_each do |x|
+        count << true if yield(x) == true
+      end
+    elsif arg.nil?
+      count = my_select { |x| !x.nil? && x != false }
+    elsif arg.class == Class
+      count = my_select { |x| x.is_a?(arg) }
+    elsif arg.class == Regexp
+      num = 0
+      my_each do |x|
+        num += 1 if x.match(arg)
+      end
+      return num.zero?
+    end
+
+    count.size.zero?
+  end
+
+  def my_count(arg = nil)
```

Comment:
#### REQUIRED

There is a :bug: with `my_count` method. Please, let's fix this.

```ruby
$ irb
irb(main):001:0> require_relative 'lib/main'
=> true
irb(main):002:0> [1,2,3,4,5].count(4)
=> 1
irb(main):003:0> [1,2,3,4,5].my_count(4)
=> nil
irb(main):004:0> [1,2,3,4,5].count{|i| i > 3}
=> 2
irb(main):005:0> [1,2,3,4,5].my_count{|i| i > 3}
=> 5
```

Reply:
Thanks, boss and this has been resolved.